packages/flutter/lib/src/material/app_bar.dart

```diff
@@ -67,17 +67,20 @@ class _ToolbarContainerLayout extends SingleChildLayoutDelegate {
 /// app bar, see [SliverAppBar], which embeds an [AppBar] in a sliver for use in
 /// a [CustomScrollView].
 ///
-/// When not used as [Scaffold.appBar], or when wrapped in a [Hero], place the app
-/// bar in a [MediaQuery] to take care of the padding around the content of the
-/// app bar if needed, as the padding will not be handled by [Scaffold].
-///
 /// The AppBar displays the toolbar widgets, [leading], [title], and [actions],
 /// above the [bottom] (if any). The [bottom] is usually used for a [TabBar]. If
 /// a [flexibleSpace] widget is specified then it is stacked behind the toolbar
 /// and the bottom widget. The following diagram shows where each of these slots
 /// appears in the toolbar when the writing language is left-to-right (e.g.
 /// English):
 ///
+/// The [AppBar] insets its content based on the ambient [MediaQuery]'s padding,
+/// to avoid system UI intrusions. It's taken care of by [Scaffold] when used in
+/// the [Scaffold.appBar] property. When animating an [AppBar], unexpected
+/// [MediaQuery] changes (as is common in [Hero] animations) may cause the content
+/// to suddenly jump. Wrap the [AppBar] in a [MediaQuery] widget with consistent
```

Comment:
What does `consistent padding values` mean?

Reply:
`Wrap the [AppBar] in a [MediaQuery] widget, adjust its padding such that the animation is smooth`?

Reply:
This implies that there isn't a recommended practice, it varies from case to case, is this right?

Reply:
Yes that's right. For example for a bottom `AppBar`, the top padding is more likely the culprit of a broken animation.

Reply:
So I guess it's alright as the message conveys it's intent.